src/Sentry/Protocol/Envelopes/Envelope.cs

```diff
@@ -43,12 +44,54 @@ public Envelope(IReadOnlyDictionary<string, object?> header, IReadOnlyList<Envel
         /// <summary>
         /// Attempts to extract the value of "event_id" header if it's present.
         /// </summary>
-        public SentryId? TryGetEventId() =>
-            Header.TryGetValue(EventIdKey, out var value) &&
-            value is string valueString &&
-            Guid.TryParse(valueString, out var guid)
-                ? new SentryId(guid)
-                : null;
+        [Obsolete]
+        public SentryId? TryGetEventId()
+        {
+            return TryGetEventId(null);
+        }
```

Comment:
```suggestion
        public SentryId? TryGetEventId()
        {
            var logger = SentrySdk.CurrentOptions?.DiagnosticLogger;
            return TryGetEventId(logger);
        }
```

(no need to obsolete)